dbms/src/TestUtils/FunctionTestUtils.cpp

```diff
@@ -108,8 +108,10 @@ void blockEqual(
     const Block & actual)
 {
     size_t columns = actual.columns();
+    size_t expected_columns = expected.columns();
 
-    ASSERT_TRUE(expected.columns() == columns);
+    ASSERT_TRUE(expected_columns == columns)
+        << fmt::format("Expected: {}, Actual: {}", expected_columns, columns);
```

Comment:
```suggestion
    ASSERT_EQ(expected_columns, columns);
```
Use `ASSERT_EQ` and it will show the actual number when they don't match. And we don't need to manually write the message to show the value.

Reply:
Maybe we can use something like

```cpp
::testing::AssertionResult blockEqual(
    const Block & expected,
    const Block & actual)
{
    size_t columns = actual.columns();

    ASSERT_EQUAL(expected.columns(), columns, "Block size mismatch");

    for (size_t i = 0; i < columns; ++i)
    {
        const auto & expected_col = expected.getByPosition(i);
        const auto & actual_col = actual.getByPosition(i);
        auto ret = columnEqual(expected_col, actual_col);
        if (!ret)
            return ret;
    }

    return ::testing::AssertionSuccess();
}
```

Reply:
> Maybe we can use something like
> 
> ```c++
> ::testing::AssertionResult blockEqual(
>     const Block & expected,
>     const Block & actual)
> {
>     size_t columns = actual.columns();
> 
>     ASSERT_EQUAL(expected.columns(), columns, "Block size mismatch");
> 
>     for (size_t i = 0; i < columns; ++i)
>     {
>         const auto & expected_col = expected.getByPosition(i);
>         const auto & actual_col = actual.getByPosition(i);
>         auto ret = columnEqual(expected_col, actual_col);
>         if (!ret)
>             return ret;
>     }
> 
>     return ::testing::AssertionSuccess();
> }
> ```

it seems to same as https://github.com/pingcap/tiflash/blob/73e708cd22b935ca240a236a87e261aabddd770e/dbms/src/TestUtils/FunctionTestUtils.cpp#L106-L121

Reply:
Oh yeah, it is just a little refactoring. The advantage might lies in we have better output error message when column size is not expected.